File: HumanBenchmark/typingTest.js
const puppeteer = require('puppeteer');

(async () => {
  // Launch a non-headless Chrome browser
  const browser = await puppeteer.launch({ headless: false });

  // Open a new tab and navigate to the URL
  const page = await browser.newPage();
  await page.goto('https://humanbenchmark.com/tests/typing');
  await page.waitForTimeout(10000); // Wait for the page to load (adjust the time as needed)

  // Extract the paragraph text
  const paragraph = await page.evaluate(() => {
    const incompleteSpans = document.querySelectorAll('.incomplete, .incomplete.current');
    let text = '';
    incompleteSpans.forEach((span) => {
      text += span.textContent;
    });
    return text;
  });

  console.log(paragraph);

  // Type the paragraph using keyboard emulation
  await page.keyboard.type(paragraph);

  // Wait for user input to close the browser
  await page.waitForSelector('body'); // You can replace this with a specific selector if needed
  await page.waitForTimeout(10000); // Adjust the time as needed
  await browser.close();
})();

File: HumanBenchmark/typingTest.py
from selenium import webdriver
from pynput.keyboard import Key, Controller as KeyboardController
import time

keyboard = KeyboardController()

driver = webdriver.Chrome()

# Open a new tab and navigate to the URL
driver.execute_script("window.open('https://humanbenchmark.com/tests/typing', '_blank')")
driver.switch_to.window(driver.window_handles[-1])
driver.maximize_window()

# Wait for the page to load
time.sleep(10)

# Get the source code of the page
html = driver.page_source

# Find the paragraph
first = html.find('<span class="incomplete current">')
last = html.rfind('<span class="incomplete">')

# Parse the paragraph
paragraph = html[first:last+26].replace('<span class="incomplete current">', '').replace('<span class="incomplete">', '').replace('</span>', '')

print(paragraph)
print('\n')

keyboard.type(paragraph)

# Close the tab and the web driver
a = input('Apreta enter para terminar: ')
driver.close()
driver.quit()